xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -511,6 +522,7 @@ private void updateRoutes(List<VirtualHost> virtualHosts) {
       // Make newly added clusters selectable by config selector and deleted clusters no longer
       // selectable.
       currentRoutes = routes;
+      fallbackTimeoutNano = httpMaxStreamDurationNano;
```

Comment:
Note there is a race between updating `currentRoutes` and `fallbackTimeoutNano`. The config selector might select a route from the updated `currentRoutes` list while using the old `fallbackTimeoutNano`. That is, these two updates should be atomic but actually it is not. The race window should be extremely small (should be almost negligible).

One option is to create a data structure that groups these two fields so that the update can be atomic. But given the race window is small, may not be necessary to do that.

Reply:
Alright, eliminated the race by grouping the list of routes and fallback timeout value into `RoutingConfig`.